### PR TITLE
tests-common: Handle SSHConsoleConn being closed prematurely

### PR DESCRIPTION
**Pull request by matthew-soulsby**
### Description of PR

There was an issue whereby certain dut_console tests would throw an `OSError: Socket is closed` prior to the test cases being run on certain devices. 

This was caused due to the way in which, upon a successful login to the in-use console port, the device would close the connection with a `Port is in use. Closing connection...` message. 

As such, this PR adds a check during session preparation, throwing a more descriptive error in the instance a console port is blocked, so that it is easy to diagnose and resolve the issue for an affected device.

Summary:
Adds more descriptive error for a blocked console port (for dut_console tests)

### Type of change

- [ ] Bug fix
- [x] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

### Back port request
- [ ] 202012
- [ ] 202205
- [ ] 202305
- [x] 202311
- [x] 202405

#### What is the motivation for this PR?
dut_console/test_idle_timeout was failing due to the console port being blocked (on a specific device with a blocked port), but the error was a generic OSError.

#### How did you do it?
Added a more descriptive error for ease of diagnosing and resolving the issue.

#### How did you verify/test it?
Ran on the affected testbed, resulting in the new error being thrown when the device's console port is blocked.

#### Any platform specific information?
There may be platforms which do not use the same `Port is in use. Closing connection...` message - however if such a case is encountered, it would be trivial to enhance the check to account for platform differences.

### Documentation
This is a minor enhancement, so I have not updated any documentation.

### CODE REVIEW

File: tests/common/connections/ssh_console_conn.py [hunk 1]
```diff
@@ -26,7 +27,13 @@ def __init__(self, **kwargs):
         super(SSHConsoleConn, self).__init__(**kwargs)
 
     def session_preparation(self):
-        self._test_channel_read()
+        session_init_msg = self._test_channel_read()
+        self.logger.debug(session_init_msg)
+
+        if re.search(r"Port is in use. Closing connection...", session_init_msg, flags=re.M):
+            console_port = self.username.split(':')[-1]
+            raise PortInUseException(f"Host closed connection, as console port '{console_port}' is currently occupied.")
```
Comment (ZhaohuiS) [inline tests/common/connections/ssh_console_conn.py:35]: @matthew-soulsby what's the result for case dut_console/test_idle_timeout if this exception happens? Will it capture the exception and give a more specific failed reason with pytest.fail?

Comment (matthew-soulsby) [reply tests/common/connections/ssh_console_conn.py:35]: @ZhaohuiS So the exception is left uncaptured, and the way that it is displayed is as follows (extracted from the test run xml):
```xml
<error message="failed on setup with `tests.common.connections.ssh_console_conn.PortInUseException: Host aborted connection, as console port '36' is currently occupied.`">
```

All of this executes in the setup [here](https://github.com/sonic-net/sonic-mgmt/blob/9e3cc9b8394adf8ce9fa2215adec4f17d0cb7e27/tests/conftest.py#L1733), prior to the test case itself being run, so I don't believe that pytest.fail would work here, but the exception thrown provides useful information (regarding the device, port and the port being in use), along with the fact that it was the setup that failed.

Happy to make changes if there's a better way of doing it though!

**Review by ZhaohuiS — APPROVED** [review]
LGTM

**Review by StormLiangMS — APPROVED** [review]
LGTM

**Pull request merged**